app/models/comment.rb

```diff
@@ -1,4 +1,11 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
+
+  after_create :update_post_comments_counter
+
+  # Method to update the comments counter for a post
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
- [ ] Kindly I recommend you to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #`update_post_comments_counter `method every time a new post has been saved to database, then make the update_post_comments_counter` method private.
- `after_create` is a callback that is only triggered when a new object is created. It will not be triggered when an object is updated.

- `after_save`, on the other hand, is triggered every time an object is saved. This means it will be called both when an object is created and when it is updated.

- [x] In your case, if you use `after_create` :`update_post_comments_counter`, the `update_post_comments_counter` method will only be called when a new comment is created. If a comment is updated, the counter will not change.

- [x] If you use` after_save :update_post_comments_counter,` the `update_post_comments_counter `method will be called every time a comment is created or updated. This might be useful if the update can change whether the comment is counted or not. So kindly I recommend you to use **` after_save :update_post_comments_counter,`**

 ```
after_save : update_post_comments_counter

  private

  def update_post_comments_counter
    post.update(comments_counter: post.comments.count)
  end
```